docs/bel_and_site_design.md

```diff
@@ -0,0 +1,132 @@
+## Cell, BEL and Site Design
+
+One of the key concepts within the FPGA interchange device resources is the
+relationship between the cell library and the device BEL and site definitions.
+A well designed cell library and a flexible but consise BEL and site
+definition is important for exposing the hardware in an efficient way that
+enables a place and route tool to succeed.
+
+Good design is hard to capture, but this document will talk about some of the
+considerations.
+
+## Granularity of the cell library
+
+It is important to divide the place and route problem and the synthesis
+problem, at least as defined for the purpose of the FPGA interchange.  The
+synthesis tool operates on the **cell library**, which should be designed to
+expose logic elements at a useful level of granularity.
+
+As a concrete example, a LUT4 element is techinically just two LUT3 elements,
+connected by a mux (e.g. MUXF4), a LUT3 element is just two LUT2 elements,
+connected by a mux (e.g. MUXF3), etc. If the outputs of those interior muxes
+are not accessible to the place and route tool, then exposing those interior
+function muxes as cells in the cell library is not a useful.
+
+Cell definitions should be granular enough that the synthesis can map to
+them, but not so granular that the place and route tool will be making few if
+any choices.  If there is only one legal placement of the cell, it's value is
+relatively low.
+
+## Drawing site boundries
+
+When designing an FPGA interchange device resource for a new fabric, one
+important consideration is where to draw the site boundary.  The primary goal
+of lumping BELs within a site is to capture some local congestion due to
+fanout limitations.  Interior static routing muxes and output muxes may
+accomidate significantly fewer signals than the possible number of BELs that
+drive them.  In this case, it is important to draw the site boundary large
+enough to capture these cases so as to enable the local congestion to be
+resolved during either packing for clustered approaches, or during placement
+during unclustered approaches.  In either case, local congestion that is
+strongly placement dependant must be resolved prior to general routing,
+unless a fused placement and routing algorithm is used.
+
+## Drawing BEL boundaries
+
+BEL definitions require that creating a boundary around primitive elements of
```

Comment:
Probably something is missing from this sentence

Reply:
Fixed